relays/messages/src/message_race_delivery.rs

```diff
@@ -652,73 +641,27 @@ async fn select_nonces_for_delivery_transaction<P: MessageLane>(
 			DispatchFeePayment::AtTargetChain => selected_unpaid_weight,
 		};
 
-		// now the message has passed all 'strong' checks, and we CAN deliver it. But do we WANT
-		// to deliver it? It depends on the relayer strategy.
-		match relayer_mode {
-			RelayerMode::Altruistic => {
-				soft_selected_count = index + 1;
-			},
-			RelayerMode::Rational => {
-				let delivery_transaction_cost = lane_target_client
-					.estimate_delivery_transaction_in_source_tokens(
-						hard_selected_begin_nonce..=
-							(hard_selected_begin_nonce + index as MessageNonce),
-						new_selected_prepaid_nonces,
-						new_selected_unpaid_weight,
-						new_selected_size as u32,
-					)
-					.await
-					.map_err(|err| {
-						log::debug!(
-							target: "bridge",
-							"Failed to estimate delivery transaction cost: {:?}. No nonces selected for delivery",
-							err,
-						);
-					})
-					.ok()?;
-
-				// if it is the first message that makes reward less than cost, let's log it
-				// if this message makes batch profitable again, let's log it
-				let is_total_reward_less_than_cost = total_reward < total_cost;
-				let prev_total_cost = total_cost;
-				let prev_total_reward = total_reward;
-				total_confirmations_cost =
```

Comment:
Previous code:
```
let total_confirmations_cost = 0;
for _ in all_ready_nonces {
...
  total_confirmations_cost = total_confirmations_cost.saturating_add(&confirmation_transaction_cost);
...
}
```

New code:
```
for _ in all_ready_nonces {
...
  let total_confirmations_cost = 0;
  total_confirmations_cost = total_confirmations_cost.saturating_add(&confirmation_transaction_cost);
...
}
```

That's wrong. The same is for other nearby variables